### PR TITLE
Updates target locator test + W3c to pass on Firefox. Updates documentation to reflect common functionality.

### PR DESCRIPTION
**Pull request by staats-google**

### CODE REVIEW

File: lib/src/sync/common.dart [hunk 1]
```diff
@@ -16,6 +16,7 @@ import 'web_driver.dart';
 import 'web_element.dart';
 
 const String elementStr = 'ELEMENT';
+const String w3cElementStr = 'element-6066-11e4-a52e-4f735466cecf';
```
Comment (manuelF) [inline lib/src/sync/common.dart]: What is this constant?

File: lib/src/sync/w3c_spec/target_locator.dart [hunk 2]
```diff
@@ -27,7 +27,19 @@ class W3cTargetLocator implements TargetLocator {
...
+    if (frame == null || frame is int) {
+      _resolver.post('frame', {'id': frame});
+    } else if (frame is W3cWebElement) {
+      _resolver.post('frame', {
+        'id': {w3cElementStr: frame.id}
+      });
+    } else {
+      // If you're here, you probably screwed up. But in the case someone
+      // depends on this behavior, we allow you to pass arbitrary objects.
+      _resolver.post('frame', {'id': frame});
```
Comment (manuelF) [inline lib/src/sync/w3c_spec/target_locator.dart]: Maybe a test for this too?

File: lib/src/sync/common.dart (hunk 1 above)
Comment (staats-google) [inline lib/src/sync/common.dart]: This is the "web element constant", reference here: https://www.w3.org/TR/webdriver/#elements

It replaces the use of 'ELEMENT' in the W3C spec. 

I am curious how one arrives at *this* as a magic constant in a spec, but not enough to hunt down the history here.

Comment (manuelF) [inline lib/src/sync/common.dart]: Can you leave a comment in the code right there with this URL? It's way too magic.

File: lib/src/sync/w3c_spec/target_locator.dart (hunk 2 above)
Comment (staats-google) [inline lib/src/sync/w3c_spec/target_locator.dart]: Done. Take a look, it's awkward but works for both specs.

**Comment by staats-google** [comment]
I've updated the documentation to indicate where the constants come from and renamed the JSON constant for clarity in the codebase.

**Comment by staats-google** [comment]
Tests stalled but target_locator at least passes (along with everything else).

**Pull request merged**